### PR TITLE
stop ps pod when master pod is stopped

### PR DESCRIPTION
**Pull request by QiJune**
I find that we must call `read_namespaced_pod `  inside the loop. Otherwise, we will get outdated pod status.

This PR will make ps pod stopped elegantly:

```
kubectl get pods
NAME                            READY   STATUS      RESTARTS   AGE
elasticdl-test-mnist-master     0/1     Completed   0          3m6s
elasticdl-test-mnist-ps-0       0/1     Completed   0          3m3s
elasticdl-test-mnist-worker-0   0/1     Completed   0          3m2s
```

### CODE REVIEW

File: elasticdl/python/ps/parameter_server.py [hunk 1]
```diff
@@ -29,6 +30,9 @@ def __init__(self, args):
         # Create Parameters instance
         self.parameters = Parameters()
 
+        self.master_name = "elasticdl-%s-master" % args.job_name
```
Comment (brightcoder01) [inline elasticdl/python/ps/parameter_server.py]: Can we put the naming formats in a common file. I think multiple component will use this.

Comment (terrytangyuan) [reply elasticdl/python/ps/parameter_server.py]: You can reuse `k8s_client.Client.get_master_pod_name()`

Comment (QiJune) [reply elasticdl/python/ps/parameter_server.py]: Yes, but only master has a k8s_client object.  I will make it as a common function.

File: elasticdl/python/ps/parameter_server.py [hunk 2]
```diff
@@ -63,10 +68,16 @@ def prepare(self):
...
+        config.load_incluster_config()
+        api = client.CoreV1Api()
         try:
             while True:
-                # TODO: add loop break condition
                 time.sleep(30)
+                master_pod = api.read_namespaced_pod(
+                    namespace=self.namespace, name=self.master_name
+                )
+                if master_pod.status.phase == "Succeeded":
```
Comment (skydoorkai) [inline elasticdl/python/ps/parameter_server.py:79]: If master pod status is `Failed`, ps also should stop.

Comment (QiJune) [reply elasticdl/python/ps/parameter_server.py:79]: Done

**Review by skydoorkai — APPROVED** [review]

**Pull request merged**